### PR TITLE
Refactor amplify_core to amplify_flutter

### PR DESCRIPTION
**Pull request by Amplifiyer**
Issue/RFC: https://github.com/aws-amplify/amplify-flutter/issues/263
*Description of changes:*
1. Refactor amplify_core and amplify_core_plugin_interface into one package amplify_flutter
2. Enable iOS unit tests for amplify_flutter
3. Update sample apps to use the new path
4. update plugin unit tests to not rely on the top level `Amplify` facade API.
5. various default indentation fixes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: melos.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: amplify_flutter
+name: amplify_flutter_repository
```
Comment (jamesonwilliams) [inline melos.yaml:1]: Why this change?

File: packages/amplify_auth_cognito/example/lib/Widgets/ConfirmResetWidget.dart [hunk 2]
```diff
@@ -1,33 +1,31 @@
 import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
```
Comment (jamesonwilliams) [inline packages/amplify_auth_cognito/example/lib/Widgets/ConfirmResetWidget.dart:1]: There are some weird things happening with indentation in this file. While I was authoring Dart in VS Code recently, I had a fairly bad experience with its automatic formatting. We probably need to get some team wide standards in place. A very basic one: is the indent level 2 or 4 or 8? Right now, I think there are examples across Dart/Kotlin/Swift files of each. I've _assumed_ that 2 is correct for Dart files, and 4 is correct for Kotlin files. But, there are currently many counter examples where Dart has 4 and Kotlin has 2.

File: packages/amplify_flutter/android/src/test/kotlin/com/amazonaws/amplify/AmplifyTest.kt [hunk 3]
```diff
@@ -13,15 +13,15 @@ import org.robolectric.RobolectricTestRunner
 
 
 @RunWith(RobolectricTestRunner::class)
-class AmplifyCoreTest {
+class AmplifyTest {
 
-  lateinit var plugin: Core
+  lateinit var plugin: Amplify
```
Comment (jamesonwilliams) [inline packages/amplify_flutter/android/src/test/kotlin/com/amazonaws/amplify/AmplifyTest.kt:18]: Should probably change the variable name here, too. `amplify`? `fascade`?
```suggestion
  lateinit var fascade: Amplify
```

File: packages/amplify_flutter/example/README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
-# amplify_core_example
+# amplify_flutter_example
 
-Demonstrates how to use the amplify_core plugin.
+Demonstrates how to use amplify.
```
Comment (jamesonwilliams) [inline packages/amplify_flutter/example/README.md:3]: Capitalized for proper noun:
```suggestion
Demonstrates how to use Amplify.
```

File: packages/amplify_flutter/pubspec.yaml [hunk 5]
```diff
@@ -10,7 +10,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  amplify_core_plugin_interface: ^0.0.1-dev.4
+  plugin_platform_interface: ^1.0.1
```
Comment (jamesonwilliams) [inline packages/amplify_flutter/pubspec.yaml:13]: Hm, is this a Flutter framework thing? Why this change?

**Review by jamesonwilliams — APPROVED** [review]
Cool! So, this looks like the first of a series of changes, right? Here, we're renaming the module. Next, we'll start building out the "bottom layer" of re-usable dependency types.

File: packages/amplify_flutter/pubspec.yaml (hunk 5 above)
Comment (Amplifiyer) [reply packages/amplify_flutter/pubspec.yaml:13]: yes, this is because of combining the `amplify_core_plugin_interface` package in amplify_flutter. This is to enable native platform communication which is needed for `configure()`

File: packages/amplify_auth_cognito/example/lib/Widgets/ConfirmResetWidget.dart (hunk 2 above)
Comment (Amplifiyer) [reply packages/amplify_auth_cognito/example/lib/Widgets/ConfirmResetWidget.dart:1]: You are right, dart is 2, kotlin and swift is 4

File: melos.yaml (hunk 1 above)
Comment (Amplifiyer) [reply melos.yaml:1]: This was a weird issue by melos where it would consider the melos config name as top level package and conflicts with the new amplify_flutter package.

**Comment by Amplifiyer** [comment]
> Cool! So, this looks like the first of a series of changes, right? Here, we're renaming the module. Next, we'll start building out the "bottom layer" of re-usable dependency types.

Correct, this is so that important API changes don't get lost in the ocean of refactor.

File: example/ios/Runner.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -237,15 +237,18 @@
 				"${BUILT_PRODUCTS_DIR}/AWSS3/AWSS3.framework",
 				"${BUILT_PRODUCTS_DIR}/Amplify/Amplify.framework",
 				"${BUILT_PRODUCTS_DIR}/AmplifyPlugins/AmplifyPlugins.framework",
+				"${BUILT_PRODUCTS_DIR}/AppSyncRealTimeClient/AppSyncRealTimeClient.framework",
 				"${BUILT_PRODUCTS_DIR}/DKImagePickerController/DKImagePickerController.framework",
 				"${BUILT_PRODUCTS_DIR}/DKPhotoGallery/DKPhotoGallery.framework",
 				"${PODS_ROOT}/../Flutter/Flutter.framework",
 				"${BUILT_PRODUCTS_DIR}/ObjectMapper/ObjectMapper.framework",
+				"${BUILT_PRODUCTS_DIR}/ReachabilitySwift/Reachability.framework",
```
Comment (haverchuck) [inline example/ios/Runner.xcodeproj/project.pbxproj:245]: Do we know what exactly this Reachability library is doing here?  This was related to the dependency conflict that amplify-ios was workign on?

Comment (Amplifiyer) [reply example/ios/Runner.xcodeproj/project.pbxproj:245]: This is auto generated by the xCode and not manually changed. I'm guessing with new version of amplify-iOS library this will go away?

File: example/pubspec.yaml [hunk 7]
```diff
@@ -25,7 +25,7 @@ dependencies:
   flutter:
     sdk: flutter
   file_picker: ^1.8.0+1
-  amplify_core: ^0.0.1-dev.4
+  amplify_flutter: ^0.0.1-dev.4
```
Comment (haverchuck) [inline example/pubspec.yaml:28]: It feels a little weird to roll out a "new" package with this version, but I suppose if we are keeping all these numbers in sync it makes sense.

File: packages/amplify_analytics_pinpoint/example/lib/main.dart [hunk 8]
```diff
@@ -129,7 +129,7 @@ class _MyAppState extends State<MyApp> {
     return MaterialApp(
       home: Scaffold(
           appBar: AppBar(
-            title: const Text('Amplify Core example app'),
+            title: const Text('Amplify Analytics example app'),
```
Comment (haverchuck) [inline packages/amplify_analytics_pinpoint/example/lib/main.dart:132]: Nice catch.

**Review by haverchuck — APPROVED** [review]
I left a couple of minor questions, but I'm still approving it.

File: melos.yaml (hunk 1 above)
Comment (Salakar) [reply melos.yaml:1]: If you can would you mind raising a issue over at https://github.com/invertase/melos so I can take a look - doesn't seem right that this should be causing any conflicts but I could be wrong also 😅

Comment (Amplifiyer) [reply melos.yaml:1]: @Salakar https://github.com/invertase/melos/issues/47
